cmd/mailer/main.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/fdemchenko/exchanger/internal/communication/mailer"
+	amqp "github.com/rabbitmq/amqp091-go"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	DefaultSMTPPort                 = 25
+	DefaultRabbitMQPort             = 5672
+	DefaultMailerConnectionPoolSize = 3
+)
+
+func main() {
+	conn, err := amqp.Dial(ServiceConfig.RabbitMQConnString)
+	if err != nil {
+		log.Fatal().Err(err).Send()
+	}
+	ch, err := conn.Channel()
+	if err != nil {
+		log.Fatal().Err(err).Send()
+	}
+
+	queue, err := ch.QueueDeclare(
+		"emails",
+		false,
+		false,
+		false,
+		false,
+		nil,
+	)
+	if err != nil {
+		log.Fatal().Err(err).Send()
+	}
+
+	deliveries, err := ch.Consume(queue.Name, "", true, false, false, false, nil)
+	if err != nil {
+		log.Fatal().Err(err).Send()
+	}
+	log.Info().Str("name", queue.Name).Msg("Mialer service started")
+
+	mailerService := NewMailerService(ServiceConfig.SMTP)
+	mailerService.StartWorkers(ServiceConfig.SMTP.ConnectionPoolSize)
+
+	for delivery := range deliveries {
+		message := mailer.Message[json.RawMessage]{}
+		err := json.Unmarshal(delivery.Body, &message)
+		if err != nil {
+			log.Error().Err(err).Send()
+			continue
+		}
+
+		log.Debug().Str("message type", message.Type.String()).
+			Time("timestamp", message.Timestamp).
+			Msg("New message received")
+
+		ParseMessageType(mailerService, message)
```

Comment:
Looks like this must be a method of `mailerService`

Reply:
Yeah, agree! I've created a function with explicit pointer receiver and didn't managed to make it method of service. What's a bad thing to write code in late night:)